apps/labs/posts/code-generator-and-nebari.md

```diff
@@ -0,0 +1,159 @@
+
+To explain more about how the generates templates, let's assume I made some files like `main.py`, `model.py`, and `utils.py` as a template. Now we use the [ejs](https://ejs.co) project to design the templates. To illustrate this, let us take an example from the templates.
+
+```python
+#::: if ((it.argparser == 'fire')) { :::#
+    fire.Fire(main)
+#::: } else if ((it.argparser == 'hydra')){ :::#
+    sys.argv.append("hydra.run.dir=.")
+    sys.argv.append("hydra.output_subdir=null")
+    sys.argv.append("hydra/job_logging=stdout")
+    main()
+#::: } else { :::#
+    main()
+#::: } :::#
+```
+
+Now, as you can see, there are some commented codes. This is part of JS, which helps select different configurations of [argparsers](http://argparsers.It). Seems easy, right?
+
+But it can be challenging to manage, and we also need to ensure we satisfy the `lint` formatting for the CI (which can be pretty tricky with templates :-( ). Still, these JS features can be very powerful, and we are trying to improve these as much as possible. By the way, if you are confused about where this `it.argparser` selector came from, better check the project, but the short answer is it comes from a `metadata.json` file that maintains the options for all the templates in the app.
+
+Some of the related PRs can be found here: [#288](https://github.com/pytorch-ignite/code-generator/pull/288) [#283](https://github.com/pytorch-ignite/code-generator/pull/283)
+
+# The Way Forward for the Project
+
+The Code generator project seems to be moving forward at a great pace and is expected to have many more additions in the future. Starting and maintaining a new project is challenging in open source and can have a very high risk/reward ratio. Still, good communities try their best to make and maintain projects that can be helpful for the maximum number of people and reduce the friction for new people entering the community. Some of the issues I suggest and may work on in the future are
+
+### 1. Separate `metadata.json` for each template
+
+We are considering providing separate metadata.json files for each template or something similar to that in the future. I proposed an issue to the project for the same [here](https://github.com/pytorch-ignite/code-generator/issues/308) and will try to work on this in the future.
+
+### 2. Script to contribute new templates
+
+Since it is hard for Data science enthusiasts to learn JS and contribute templates easily, which also seems to be a point of friction in adopting the app, we proposed adding a script that can help make J.S. and CI-related changes to the app as suggested in the issue [here](https://github.com/pytorch-ignite/code-generator/issues/316). This issue may take a bit of hard work, but it can be a great addition to the app.
```

Comment:
```suggestion
Since it is hard for Data science enthusiasts to learn JS and contribute templates easily, which also seems to be a point of friction in adopting the app, we proposed adding a script that can help make JS and CI-related changes to the app as suggested in the issue [here](https://github.com/pytorch-ignite/code-generator/issues/316). This issue may take a bit of hard work, but it can be a great addition to the app.
```